include/pagmo/problems/translate.hpp

```diff
@@ -140,43 +181,122 @@ class translate : public problem
+     * The has_gradient_sparsity computation is forwarded to the inner problem.
+     *
+     * @return a flag signalling the availability of the gradient sparisty in the inner problem.
+     */
+    bool has_gradient_sparsity() const
+    {
+        return m_problem.has_gradient_sparsity();
+    }
+
+    /// Gradient sparsity.
+    /**
+     * The gradient sparsity computation is forwarded to the inner problem.
+     *
+     * @return the gradient sparsity of the inner problem.
+     */
+    sparsity_pattern gradient_sparsity() const
+    {
+        return m_problem.gradient_sparsity();
+    }
+
+    /// Checks if the inner problem has hessians.
+    /**
+     * The has_hessians computation is forwarded to the inner problem.
+     *
+     * @return a flag signalling the availability of the hessians in the inner problem.
+     */
+    bool has_hessians() const
+    {
+        return m_problem.has_hessians();
+    }
+
+    /// Hessians.
+    /**
+     * The hessians computation is forwarded to the inner problem, after the translation of \p x.
      *
      * @param x the decision vector.
      *
-     * @return the hessians of the fitness function.
+     * @return the hessians of the fitness function computed at \p x.
      *
      * @throws unspecified any exception thrown by memory errors in standard containers,
      * or by problem::hessians().
      */
     std::vector<vector_double> hessians(const vector_double &x) const
     {
         vector_double x_deshifted = translate_back(x);
-        return static_cast<const problem *>(this)->hessians(x_deshifted);
+        return m_problem.hessians(x_deshifted);
+    }
+
+    /// Checks if the inner problem has hessians sparisty implemented.
+    /**
+     * The has_hessians_sparsity computation is forwarded to the inner problem.
+     *
+     * @return a flag signalling the availability of the hessians sparisty in the inner problem.
+     */
+    bool has_hessians_sparsity() const
+    {
+        return m_problem.has_hessians_sparsity();
+    }
+
+    /// Hessians sparsity.
+    /**
+     * The hessians sparsity computation is forwarded to the inner problem.
+     *
+     * @return the hessians sparsity of the inner problem.
+     */
+    std::vector<sparsity_pattern> hessians_sparsity() const
+    {
+        return m_problem.hessians_sparsity();
+    }
+
+    /// Calls <tt>has_set_seed()</tt> of the inner problem.
+    /**
+     * Calls the method <tt>has_set_seed()</tt> of the inner problem.
+     *
+     * @return a flag signalling wether the inner problem is stochastic.
+     */
+    bool has_set_seed() const
+    {
+        return m_problem.has_set_seed();
+    }
+
+    /// Calls <tt>set_seed()</tt> of the inner problem.
+    /**
+     * Calls the method <tt>set_seed()</tt> of the inner problem.
+     *
+     * @param seed seed to be set.
+     *
+     * @throws std::not_implemented_error if the inner problem is not stochastic.
```

Comment:
This should be ``@throws unspecified any exception thrown by the method <tt>set_seed()</tt> of the inner problem.``. ``not_implemented_error`` is our own exception, not part of ``std``.